proposals/stdlib/instant.md

```diff
@@ -0,0 +1,1420 @@
+Instant and Clock.
+=================
+
+* **Type**: Standard Library API proposal
+* **Author**: Dmitry Khalanskiy
+* **Contributors**: Ilya Gorbunov, Roman Elizarov, Vsevolod Tolstopyatov
+* **Status**: pending
+* **Prototype**: available in `kotlinx-datetime`
+
+Summary
+-------
+
+This proposal aims to introduce facilities for obtaining the current system time
+into the Kotlin Standard Library.
+
+Additionally, it defines some interactions with `kotlin.time.Duration`, an
+existing API for representing some number of seconds.
+
+The API described here is available in `kotlinx-datetime` 0.6.0 in the form of
+the `Instant` class and the `Clock` interface.
+The goal of this proposal is to move this widely applicable functionality into
+the Standard Library.
+
+Overview
+--------
+
+### `Instant`
+
+`Instant` denotes a moment in time, regardless of the observer.
```

Comment:
Maybe best to remove the phrase "regardless of observer" since that evokes relativistic effects.  According to the theory of relativity, simultaneity is relative.  If events occur in different places, there may be frames of reference in which they are simultaneous, other frames in which one occurs first and other frames in which the other occurred first.

There is no "absolute" frame of reference from which we can say that two events occurring at different places occurred at the "same instant".

So, I'm thinking it's probably best not to mention "observers" since none of our usual language and intuition about instants or clocks works when we try to account for observers in different reference frames.

Reply:
Doesn't "regardless of the observer" strongly imply that relativistic effects are not accounted for, given that this definition is absurd in the context of relativity?

Reply:
"Regardless of observer" means "for any observer."  Or "in any frame of reference."

But everything else we're talking about here is only true if everyone is in the same frame of reference.

In other words, only for specific observers.

So using the phrase "regardless of observer" is the opposite of what we want.

Probably best just not to mention observers.